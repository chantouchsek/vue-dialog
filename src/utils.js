export function destroyVueElement (vm) {
  if (vm && !vm._isDestroyed && (typeof vm.$destroy === 'function')) {
    vm.$destroy()
  }
}

export function findContainer (container) {
  let found
  if (typeof container === 'string') {
    found = document.querySelector(container)
  } else {
    found = container
  }
  if (!found) {
    found = document.body
  }
  return found
}

// todo
// export function middlewareSeries (promises, appContext) {
//   if (!promises.length || appContext._redirected || appContext._errored) {
//     return Promise.resolve()
//   }
//   return promisify(promises[0], appContext)
//     .then(() => {
//       return middlewareSeries(promises.slice(1), appContext)
//     })
// }
