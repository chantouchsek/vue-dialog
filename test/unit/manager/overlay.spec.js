import DialogManager from '../../../src/manager'
import Vue from 'vue'
import { sleep } from '../../utils'

describe('manager', () => {
  let manager

  const overlay = {
    template: '<div v-if="visible" id="overlay"/>',
    data () {
      return {
        visible: false
      }
    }
  }

  test('Check overlay correct work', async () => {
    manager = new DialogManager({ context: { store: {} } })
    manager.overlay('default', overlay)
    let dialog = manager.show({
      template: '<p/>',
      asyncData: sleep(10)
    })
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
    dialog = await dialog
    expect(dialog.element).toMatchSnapshot()
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
    manager.overlay('default').destroy()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check overlay with extended component', async () => {
    const dlg = Vue.extend({
      template: '<p/>',
      asyncData: sleep(5),
      overlay: 'extended'
    })

    const overlay = Vue.extend({
      template: '<div v-if="visible" id="overlay">extended</div>',
      data () {
        return {
          visible: false
        }
      }
    })
    manager = new DialogManager({})
    manager.overlay('extended', overlay)
    let dialog = manager.show(dlg)
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
    dialog = await dialog
    expect(dialog.element).toMatchSnapshot()
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
  })
})
