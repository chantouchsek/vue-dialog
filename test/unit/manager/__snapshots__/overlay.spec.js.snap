// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Check overlay correct work 1`] = `"<div id=\\"overlay\\"></div>"`;

exports[`manager Check overlay correct work 2`] = `
<div
  class="dialog-layout"
>
  <p />
</div>
`;

exports[`manager Check overlay correct work 3`] = `"<!---->"`;

exports[`manager Check overlay with extended component 1`] = `"<div id=\\"overlay\\">extended</div>"`;

exports[`manager Check overlay with extended component 2`] = `
<div
  class="dialog-layout"
>
  <p />
</div>
`;

exports[`manager Check overlay with extended component 3`] = `"<!---->"`;
